tests/b2share_unit_tests/files/test_b2share_storage.py

```diff
@@ -214,3 +216,121 @@ def test_embargoed_records_with_external_pids(app, test_users,
                                        key=file.obj.key)
                 resp = client.get(ext_file_url)
                 assert resp.status_code == 302
+
+
+def test_generation_external_pids(app, records_data_with_external_pids,
+                                  deposit_with_external_pids):
+    """Test the generate_external_pids function."""
+    expected_output = records_data_with_external_pids['external_pids']
```

Comment:
Maybe we could put the PIDs in reverse order in `records_data_with_external_pids['external_pids']` just to check that  `generate_external_pids` put them back in alphabetical order. Or add a PID at the wrong position.